Source/NSURL.m

```diff
@@ -92,15 +85,12 @@
 
 #import "GNUstepBase/NSURL+GNUstepBase.h"
 
-#define	GSInternal NSURLComponentsInternal
-#include "GSInternal.h"
-GS_PRIVATE_INTERNAL(NSURLComponents)
-
 NSString * const NSURLErrorDomain = @"NSURLErrorDomain";
 NSString * const NSErrorFailingURLStringKey = @"NSErrorFailingURLStringKey";
 
 @interface	NSString (NSURLPrivate)
 - (NSString*) _stringByAddingPercentEscapes;
```

Comment:
We should replace these two methods by proper NSCharactersets for the encoding and call stringByAddingPercentEncodingWithAllowedCharacters: with these.
@rfm could you help with setting up these?

Reply:
Agreed